src/zrc/api/tests/test_audittrails.py

```diff
@@ -0,0 +1,198 @@
+import unittest
+from copy import deepcopy
+
+from django.test import override_settings
+
+from rest_framework.test import APITestCase
+from vng_api_common.constants import VertrouwelijkheidsAanduiding
+from vng_api_common.tests import JWTScopesMixin, generate_jwt, reverse
+from zds_client.tests.mocks import mock_client
+
+from zrc.datamodel.models import AuditTrail, Zaak
+from zrc.datamodel.tests.factories import StatusFactory, ZaakFactory
+from zrc.tests.constants import POLYGON_AMSTERDAM_CENTRUM
+from zrc.tests.utils import ZAAK_WRITE_KWARGS
+
+from ..scopes import (
+    SCOPE_ZAKEN_BIJWERKEN, SCOPE_ZAKEN_CREATE, SCOPE_ZAKEN_ALLES_VERWIJDEREN
+)
+
+# ZTC
+ZTC_ROOT = 'https://example.com/ztc/api/v1'
+DRC_ROOT = 'https://example.com/drc/api/v1'
+CATALOGUS = f'{ZTC_ROOT}/catalogus/878a3318-5950-4642-8715-189745f91b04'
+ZAAKTYPE = f'{CATALOGUS}/zaaktypen/283ffaf5-8470-457b-8064-90e5728f413f'
+RESULTAATTYPE = f'{ZAAKTYPE}/resultaattypen/5b348dbf-9301-410b-be9e-83723e288785'
+INFORMATIE_OBJECT = f'{DRC_ROOT}/enkelvoudiginformatieobjecten/1234'
+
+
+@override_settings(
+    LINK_FETCHER='vng_api_common.mocks.link_fetcher_200',
+    ZDS_CLIENT_CLASS='vng_api_common.mocks.MockClient'
+)
+class AuditTrailTests(JWTScopesMixin, APITestCase):
+    def setUp(self):
+        token = generate_jwt(
+            scopes=[
+                SCOPE_ZAKEN_CREATE,
+                SCOPE_ZAKEN_BIJWERKEN,
+                SCOPE_ZAKEN_ALLES_VERWIJDEREN],
+            zaaktypes=['https://example.com/zaaktype/123']
+        )
+        self.client.credentials(HTTP_AUTHORIZATION=token)
+
+        self.responses = {
+            'https://example.com/zaaktype/123': {
+                'url': 'https://example.com/zaaktype/123',
+                'productenOfDiensten': [
+                    'https://example.com/product/123',
+                    'https://example.com/dienst/123',
+                ]
+            }
+        }
+
+    def create_zaak(self):
+        url = reverse('zaak-list')
+
+        zaak_data = {
+            'zaaktype': 'https://example.com/zaaktype/123',
+            'vertrouwelijkheidaanduiding': VertrouwelijkheidsAanduiding.openbaar,
+            'bronorganisatie': '517439943',
+            'verantwoordelijkeOrganisatie': '517439943',
+            'registratiedatum': '2018-12-24',
+            'startdatum': '2018-12-24',
+            'productenOfDiensten': ['https://example.com/product/123']
+        }
+        with mock_client(self.responses):
+            response = self.client.post(url, zaak_data, **ZAAK_WRITE_KWARGS)
+
+        return response.data
+
+    def test_create_zaak_audittrail(self):
+        zaak_response = self.create_zaak()
```

Comment:
Use a factory instead?

Reply:
But, if I understand it correctly, using a factory to create a Zaak would not use the API to do so, which means that no Audittrail will be created for that operation, right?

Reply:
Woops, correct :)